features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt

```diff
@@ -79,6 +82,22 @@ internal class TextCompositionGroupMapper(
         )
     }
 
+    private fun resolveCapturedText(parameters: Sequence<ComposableParameter>, uiContext: UiContext): String {
+        val originalText = (parameters.firstOrNull { it.name == PARAM_NAME_TEXT }?.value as? String).orEmpty()
+        return when (uiContext.privacy) {
+            SessionReplayPrivacy.ALLOW -> originalText
+            SessionReplayPrivacy.MASK_USER_INPUT ->
+                if (uiContext.hasOptionSelectorParent) FIXED_INPUT_MASK else originalText
+
+            SessionReplayPrivacy.MASK ->
+                if (uiContext.hasOptionSelectorParent) {
+                    FIXED_INPUT_MASK
+                } else {
+                    StringObfuscator.getStringObfuscator().obfuscate(originalText)
+                }
```

Comment:
I know that `Spinner` or likewise components don't exist in Compose, but if I understand correctly, the first way you propose, we still need to keep this information in `UiContext` (maybe just need to change the naming). The thing I need to change is to remove the comment of "TODO" for providing `OptionSelectorDetector` for it, because we are not able to get this information with the input of `ViewGroup`, we need to have another way just like you proposed.

About the second proposition, we are just discussing "Fine Grained Masking" for Q3, it might some crossover with this idea. and that might be a good one.